src/wolframite/runtime/system.clj

```diff
@@ -0,0 +1,161 @@
+(ns wolframite.runtime.system
+  "For dealing with the runtime system, primarily the operating system specific defaults."
+  (:require
+   [babashka.fs :as fs]
+   [clojure.string :as string])
+  (:import
+   (java.lang System)))
+
+(def supported-OS #{:linux :mac :windows})
+
+;; Default path information can be listed here, in order of preference.
+(def ^:private defaults
+  [{:root  "/usr/local/Wolfram/Mathematica"
+    :kernel "Executables/MathKernel"
+    :product :mathematica
+    :os :linux}
+   {:root  "/opt/Mathematica"
+    :kernel "Executables/MathKernel"
+    :product :mathematica
+    :os :linux}
+   {:root  "/usr/local/Wolfram/WolframEngine"
+    :kernel "Executables/WolframKernel"
+    :product :wolfram-engine
+    :os :linux}
+
+   {:root  "/Applications/Mathematica.app/Contents"
+    :kernel "MacOS/MathKernel"
+    :product :mathematica
+    :os :mac}
+   {:root   "/Applications/Wolfram Engine.app/Contents/Resources/Wolfram Player.app/Contents"
+    :kernel "MacOS/WolframKernel"
+    :product :wolfram-engine
+    :os :mac}
+
+   {:root "/c:/Program Files/Wolfram Research/Mathematica/"
+    :kernel "WolframKernel.exe"
+    :product :mathematica
+    :os :windows}
+   {:root "/c:/Program Files/Wolfram Research/Wolfram Engine/"
+    :kernel "MathKernel.exe"
+    :product :wolfram-engine
+    :os :windows}])
+
+(defn- user-paths
+  "The paths given via Java property or environment variables.
+
+  NOTE: These should be root directories that distinguish Mathematica or Wolfram engine files, not subdirectories. E.g.  '/usr/local/Wolfram/Mathematica'.
+
+  TODO:
+  - (jh) alternatively, support Java system properties?"
+  []
+  (let [names  ["JLINK_JAR_PATH"
+                "MATHEMATICA_INSTALL_PATH"
+                "WOLFRAM_INSTALL_PATH"]
+        method (fn [f] (-> f (nth  2) first str (string/split #"/") last))
```

Comment:
"method" does not clarify anything. What does this fn take, why does it take its 2nd part and does all the rest? What is a sample input / output? Pls help a reader understand.